src/plugins/mer/meremulatordevice.h

```diff
@@ -35,7 +35,8 @@ class QTimer;
 QT_END_NAMESPACE
 
 namespace Mer {
-namespace Internal {
+
+using namespace Internal;
```

Comment:
This has the unwanted effect of extending Mer namespace with symbols from Mer::Internal in all modules which include this header file. I think it makes sense to move the parent class MerDevice, as well as the sibling class MerHardwareDevice out of the Internal namespace.

Reply:
If I will remove `using namespace Internal;` then I should move all classes that are used in `MerConnection` and `MerEmulatorDevice` from the `Internal` namespace. It is not only `MerDevice` and `MerHardwareDevice`, it is also `MerDeviceFactory`, `MerHardwareDeviceWidget`, `MerSdk`, `MerSdkManager`, `MerToolChain`, `MerQtVesion`, `MerVirtualBoxManager` at least.

Is that what you mean in this comment?

Reply:
Not all that are used. Just those that figure in public interfaces of the classes that you need. Unless I missed something, this is only true about MerEmulatorDeviceModel. All the other classes you named are only used in implementation and so they do not need to be exposed.